File: backend/API/serializers/nation.py
from rest_framework import serializers
from ..serializers import BaseSerializer
from ..models import Nation


class NationSerializer(BaseSerializer):
    """
    Sérializer du node Nation
    """

    name = serializers.CharField(required=True)

    # Output
    artistes = serializers.SerializerMethodField(read_only=True)

    def __init__(self, *args, **kwargs):
        super().__init__(Nation, *args, **kwargs)

    def get_artistes(self, nation):
        """
        Renvoie un lien propre vers les artistes :
        """
        return self.get_url("artiste-list", kwargs={"nation_uuid": nation.uuid})

File: frontend/src/model/nation.js
import CRUD from "./crud.js";
import Artiste from "./artiste.js";

export default class Nation extends CRUD {
    #name
    #artistes

    constructor({uuid, name, artistes}) {
        super(uuid);
        this.#name = name;
        this.#artistes = artistes;
    }

    static get endpoint() { return "nations" }

    get name() { return this.#name }
    set name(value) { this.#name = this.validateString(value, "name") }

    get artistes() { return this.fetchList(this.#artistes, Artiste) }

    toJSON() {
        return {
            uuid: this.uuid,
            name: this.#name,
        }
    }
}
